### DISCUSSION TITLE
Where are you fetching the user's avatar?

### DISCUSSION THREAD

**Discussion by zkhalapyan** [rainbow-me/rainbowkit#562] [Q&A]
My ENS avatar and the avatar presented in rainbow kit don't match, curious why it's different. Where are you fetching it? How could I let the user set their avatar? 

Thank you!

**Comment by saihaj** [comment]
> Where are you fetching it?

https://github.com/rainbow-me/rainbowkit/blob/52000e1f47ae1655b513bee18e5c137a9a2c3a16/packages/rainbowkit/src/components/ConnectButton/ConnectButtonRenderer.tsx#L83-L91

They just use `wagmi`
- https://wagmi.sh/docs/hooks/useEnsName
- https://wagmi.sh/docs/hooks/useEnsAvatar
